hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/PrefetchExecutor.java

```diff
@@ -78,7 +85,10 @@ public Thread newThread(Runnable r) {
       + Path.SEPARATOR_CHAR + ")|(" + Path.SEPARATOR_CHAR
       + HConstants.HREGION_COMPACTIONDIR_NAME.replace(".", "\\.") + Path.SEPARATOR_CHAR + ")");
 
-  public static void request(Path path, Runnable runnable) {
+  // For tests. Contains computed prefetch delay
+  private static long computedPrefetchDelay;
+
+  public static void request(Path path, boolean isInterrupted, Runnable runnable) {
```

Comment:
Why do we need the additional parameter? Just remove the future from the collection in the "interrupt" method.